app/models/page/DashboardPage.scala

```diff
@@ -88,35 +139,121 @@ object DashboardPage {
     }
   }
 
-  def fromUser(user: CachedData)(implicit request: RequestHeader, lang: Lang): DashboardPage = status(user) match {
+  private def isApplicationSubmittedAndNotWithdrawn(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    WithdrawApplicationRole.isAuthorized(user)
+
+  private def isApplicationInProgressAndNotWithdrawn(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    CreatedOrInProgressRole.isAuthorized(user)
+
+  private def isApplicationWithdrawn(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    WithdrawnApplicationRole.isAuthorized(user)
+
+  private def isApplicationCreatedOrInProgress(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    PersonalDetailsRole.isAuthorized(user)
+
+  private def isUserWithNoApplication(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    ApplicationStartRole.isAuthorized(user)
```

Comment:
@henricook @mgilgar : Agree with both... there is room for improvement here but I think it should be addressed in a separate task.